### DISCUSSION TITLE
zstd: add SpeedNoCompression

### DISCUSSION THREAD

**Discussion by dsnet** [klauspost/compress#941] [Ideas/Features]
There are situations where we must emit content as the zstd format (e.g., the protocol already negotiated ahead of time that we're operating with zstd). However, we might have some content that 1) we know won't compress well OR 2) don't want to spend CPU cycles compressing it. Could we add a `SpeedNoCompression` level that simply emits the appropriate zstd framing, but uses `Raw_Block` blocks?

This would be analogous to the existing `gzip.NoCompression` level.

**Comment by klauspost** [comment]
Sounds reasonable. I am currently afk, but I will be able to look at it next week!

**Comment by klauspost** [comment]
@dsnet 

I am considering if it would be cleaner to have a separate wrapper, but with a common interface to the Encoder.

```
type Wrapper struct {}

func NewWrapper(w io.Writer, opts ...EOption) (*Wrapper, error)

func (w *Wrapper) EncodeAll(src, dst []byte) []byte

func (w *Wrapper) Reset(w io.Writer) 

func (w *Wrapper)  ResetContentSize(w io.Writer, size int64)

func (w *Wrapper)  Write(p []byte) (n int, err error) 

... etc
```

I think I will see how the complexity/efficiency would be around adding it to the current Encoder, as that seems to be the easiest way to use it. Just want to ensure that it only keeps minimal buffers around.

(just to update that I am looking at it, but I want it done right)
